### PR TITLE
 RPM: account for nproc == 1 and update correct quadlet file

### PR DESCRIPTION
**Pull request by lsm5**
@rhatdan @dougsland @pypingou PTAL

### CODE REVIEW

**Review by dougsland — APPROVED** [review]
looks sane to me
/lgtm

File: rpm/qm.spec [hunk 1]
```diff
@@ -83,7 +83,7 @@ sed -i 's/^install: man all/install:/' Makefile
 
 # Set AllowedCPUs in quadlet file
 NPROC=$(nproc)
-if [[ $NPROC == 2 ]]; then
+if [[ $NPROC <= 2 ]]; then
```
Comment (rhatdan) [inline rpm/qm.spec:86]: I am not sure if this will work correctly, IE Will 1 work when there is only 1 cpu?  Should it be 0?

**Comment by rhatdan** [comment]
I also think this is wrong.  We should be doing
```
sed  "s/^AllowedCPUs=.*/AllowedCPUs=$ALLOWED_CPUS/" %{_datadir}/containers/systemd/qm.container > %{_sysconfdir}/containers/systemd/qm.container 
```
Currently after install I see.

 rpm -qV qm
.M.......  g /usr/lib/qm
.M.......  g /usr/lib/qm/rootfs
S.5....T.    /usr/share/containers/systemd/qm.container

File: rpm/qm.spec (hunk 1 above)
Comment (lsm5) [reply rpm/qm.spec:86]: oh you're right, my bad. sorry

**Comment by lsm5** [comment]
> I also think this is wrong. We should be doing
> 
> ```
> sed  "s/^AllowedCPUs=.*/AllowedCPUs=$ALLOWED_CPUS/" %{_datadir}/containers/systemd/qm.container > %{_sysconfdir}/containers/systemd/qm.container 
> ```
> 
> Currently after install I see.
> 
> rpm -qV qm .M....... g /usr/lib/qm .M....... g /usr/lib/qm/rootfs S.5....T. /usr/share/containers/systemd/qm.container

Remind me where the file is supposed to be? 

> I also think this is wrong. We should be doing
> 
> ```
> sed  "s/^AllowedCPUs=.*/AllowedCPUs=$ALLOWED_CPUS/" %{_datadir}/containers/systemd/qm.container > %{_sysconfdir}/containers/systemd/qm.container 
> ```
> 
> Currently after install I see.
> 
> rpm -qV qm .M....... g /usr/lib/qm .M....... g /usr/lib/qm/rootfs S.5....T. /usr/share/containers/systemd/qm.container

ack, I'll change that. Thanks

**Comment by lsm5** [comment]
@rhatdan PTAL

**Comment by lsm5** [comment]
need to fix the copr epoch issue as well.

**Comment by lsm5** [comment]
with the latest change, copr builds have the Epoch: 101 and official koji builds continue with Epoch: 0.

**Comment by rhatdan** [comment]
LGTM

File: rpm/qm.spec [hunk 2]
```diff
@@ -21,6 +29,10 @@
 %endif
 
 Name: qm
+# Set different Epochs for copr and koji
+%if %{with copr}
+Epoch: 101
+%endif
```
Comment (pypingou) [inline rpm/qm.spec:35]: Hm, what's the goal of playing with the epoch here? there are fairly high costs associated with epochs :/

**Pull request merged**